crates/metadata/src/specs.rs

```diff
@@ -1091,7 +1094,7 @@ where
 }
 
 /// The 4 byte selector to identify constructors and messages
-#[derive(Debug, Default, PartialEq, Eq, derive_more::From, JsonSchema)]
+#[derive(Debug, Default, PartialEq, Eq, derive_more::From, JsonSchema, Hash)]
```

Comment:
Shouldn't the `Hash` derive also be behind the `std` feature?

Reply:
Yes you are right. Something like this may work:
```rust
/// The 4 byte selector to identify constructors and messages
#[cfg_attr(feature = "std", derive(Hash))]
#[derive(Debug, Default, PartialEq, Eq, derive_more::From, JsonSchema)]
pub struct Selector(#[schemars(with = "String")] [u8; 4]);
...
```